File: C#/300. Longest Increasing Subsequence.cs
public class Solution {
    public int LengthOfLIS(int[] nums) {
        int n=nums.Length;
        int[] dp = new int[n+1];
        Array.Fill(dp,1);
        int res=0;
        for(int i=0;i<n;i++){
            for(int j=i-1;j>=0;j--){
                if(nums[i]>nums[j]){
                    dp[i]=Math.Max(dp[i],dp[j]+1);
                }
            }
            res=Math.Max(res,dp[i]);
        }
        return res;
    }
}

File: C++/300. Longest Increasing Subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n=nums.size();
        vector<int>dp(n+1,1);
        int res=0;
        for(int i=0;i<n;i++){
            for(int j=i-1;j>=0;j--){
                if(nums[i]>nums[j]){
                    dp[i]=max(dp[i],dp[j]+1);
                }
            }
            res=max(res,dp[i]);
        }
        return res;
    }
};
